pipeline/test/test_extract.py

```diff
@@ -139,3 +152,76 @@ def test_embedded_text_extractor(test_pdf_path):
     assert len(doc.pages[0].text_blocks[0].coords) >= 4
     assert type(doc.pages[0].text_blocks[0].coords[0]) == tuple
     assert type(doc.pages[0].text_blocks[0].coords[0][0]) == float
+
+
+def get_sample_adobe_fileref(*args, **kwargs):
+    return FileRef.create_from_local_file(
+        Path(__file__).parent / "sample_adobe_output.zip"
+    )
+
+
+def test_adobe_text_extractor(test_pdf_path, tmp_path):
+    with mock.patch.object(
+        AdobeAPIExtractor, "_get_adobe_api_result", new=get_sample_adobe_fileref
+    ):
+        data_output_dir = tmp_path / "data"
+        split_dir = tmp_path / "splits"
+        os.mkdir(data_output_dir)
+        os.mkdir(split_dir)
+
+        text_extractor = AdobeAPIExtractor(credentials_path="fake/path")
+        doc = text_extractor.extract(
+            test_pdf_path,
+            data_output_dir=data_output_dir,
+            output_folder_pdf_splits=split_dir,
+        )
+
+        assert doc.filename == test_pdf_path.name
+
+        # The returned document can have up to 8 pages (the number of pages in the original PDF).
+        # Pages with no parsed content (e.g. all figures) won't be added to the document.
+        assert len(doc.pages) <= 8
```

Comment:
For this test, can the number of pages be verified exactly?

Reply:
Alternatively I can rewrite the code so that when the Adobe API doesn't detect any text on a page, a `Page` object is still returned with `TextBlocks==[]`. Maybe that would be better in general, as then `len(doc.pages)` would always equal the number of pages in the PDF?

Reply:
I don't think a page without text blocks is useful in itself, but it'd mean that 

- `Document` has a method we can call (`len(doc.pages)`) that gives us the length of the original PDF
- we can iterate through `[page in doc.pages]` and this would match up to the pages in the existing PDF.

@chrisaballard would be interested to hear your thoughts as part of your review